x/staking/migrations/v4/migrations_test.go

```diff
@@ -0,0 +1,60 @@
+package v4_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/cosmos/cosmos-sdk/testutil"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	moduletestutil "github.com/cosmos/cosmos-sdk/types/module/testutil"
+	"github.com/cosmos/cosmos-sdk/x/distribution"
+	paramtypes "github.com/cosmos/cosmos-sdk/x/params/types"
+	"github.com/cosmos/cosmos-sdk/x/staking/migrations/v4"
+	"github.com/cosmos/cosmos-sdk/x/staking/types"
+)
+
+type mockSubspace struct {
+	ps types.Params
+}
+
+func (ms mockSubspace) HasKeyTable() bool {
```

Comment:
not sure why we need to implement these methods and/or why they panic. Can we just omit them?